crates/lib/src/ops/resolve/crate_name.rs

```diff
@@ -43,7 +43,8 @@ impl FromStr for CrateName {
 }
 
 impl CrateName {
-    pub fn dedup(mut crate_names: Vec<Self>) -> impl Iterator<Item = Self> {
+    pub fn dedup(crate_names: &[Self]) -> impl Iterator<Item = Self> {
```

Comment:
nik: Why do we switch to slice instead of `Vec` here?

Reply:
Technically, copying by ref instead of by value, but mainly means we can just borrow the caller instead of requiring a partial move out of `args` in the bin. It incurs an extra copy, but this is something called once during the entire program, on an small amount of data. So, ergonomics win out over pure efficiency.

Reply:
Thanks for the explanation!